### PR TITLE
add utility to annotate licks

### PR DESCRIPTION
**Pull request by dougollerenshaw**
Adds a function to utilities.py that will annotate the licks dataframe with some useful columns (partially stolen from @alexpiet's implementation in visual_behavior_glm (https://github.com/AllenInstitute/visual_behavior_glm/blob/86a816ebc4dcc8d450f83f77611c2ee7afdf7b93/visual_behavior_glm/GLM_fit_tools.py#L1022)

Optionally operates in place.

### Example use:
get dataset, view default licks dataframe:
```
from visual_behavior.data_access import loading
dataset = loading.get_ophys_dataset(915243090)

dataset.licks.head(10)
```
image

import `annotate_licks`:
```
from visual_behavior.utilities import annotate_licks
```

return annotated licks df as a new variable:

```
annotated_licks = annotate_licks(dataset.licks).head(10)
annotated_licks
```

image

annotate licks in place:
```
annotate_licks(dataset.licks, inplace=True)
dataset.licks.head(10)
```
image

ALSO:
fixes a bug in plotting.event_triggered_raster

### CODE REVIEW

**Comment by alexpiet** [comment]
It would be nice to finish this feature request. https://github.com/AllenInstitute/visual_behavior_analysis/issues/720

- [ ] Can you add a "bout_rewarded" label?
- [ ] Can you add a "bout_number" label so its easy to grab all the licks inside a single bout?

https://github.com/alexpiet/licking_behavior/blob/master/src/psy_metrics_tools.py#L54

**Comment by dougollerenshaw** [comment]
@alexpiet I added two columns to address your comment. I also had to change the input to the function from the licks dataframe to the entire session object so that the function would have access to both licks and rewards.

File: visual_behavior/utilities.py [hunk 1]
```diff
@@ -698,3 +698,56 @@ def event_triggered_response(df, parameter, event_times, time_key=None, t_before
         melted['event_number'] = melted['variable'].map(lambda s: s.split('event_')[1].split('_')[0])
         melted['event_time'] = melted['variable'].map(lambda s: s.split('t=')[1])
         return melted.drop(columns=['variable']).rename(columns={'value': parameter})
+
+
+def annotate_licks(dataset, inplace=False, lick_bout_ili=2):
```
Comment (alexpiet) [inline visual_behavior/utilities.py:703]: Why default the lick_bout_ili to 2 seconds? I think I default it to 750ms. I'm open to changing definitions but I think we should be consistent.

**Review by alexpiet — CHANGES_REQUESTED** [review]

**Comment by dougollerenshaw** [comment]
@matchings this conflicts with some of the work you added in PR #726. I propose that we merge this first, then strip the redundant stuff from your PR. Some comments:

* I added a boolean column called 'hit' that designates hit-licks. This aligns with the column of the same name in the df you were generating.
* I don't have a column called 'lick_in_bout', like yours. But I do have 'licks_in_bout', which is the number of licks in the bout. So to get the equivalent, I think you could just filter on 'licks_in_bout > 1'
* This is in visual_behavior.utilities. The function you generated is in visual_behavior.data_access.utilities. I can move my function if you think that's more appropriate location. Let me know in review.

File: visual_behavior/utilities.py (hunk 1 above)
Comment (dougollerenshaw) [reply visual_behavior/utilities.py]: Good point. Your default in the GLM is 0.7:
https://github.com/AllenInstitute/visual_behavior_glm/blob/86a816ebc4dcc8d450f83f77611c2ee7afdf7b93/visual_behavior_glm/GLM_params.py#L182

I matched that.

**Comment by dougollerenshaw** [comment]
@alexpiet Thanks for pointing out the fields that I missed in the docstring in your slack comment. PR is updated with missing information added.

**Review by alexpiet — APPROVED** [review]

**Pull request merged**